lib/flagsmith/engine/evaluation/mappers.rb

```diff
@@ -0,0 +1,233 @@
+        # @return [Hash] Mapped rule with :type, :conditions, and :rules
+        def self.map_segment_rule_model_to_rule(rule)
+          result = {
+            type: rule.type
+          }
+
+          # Map conditions if present
+          if rule.conditions&.any?
+            result[:conditions] = rule.conditions.map do |condition|
+              {
+                property: condition.property,
+                operator: condition.operator,
+                value: condition.value
+              }
+            end
+          else
+            result[:conditions] = []
+          end
+
+          if rule.rules&.any?
+            result[:rules] = rule.rules.map { |nested_rule| map_segment_rule_model_to_rule(nested_rule) }
+          else
+            result[:rules] = []
+          end
+
+          result
+        end
+  
+        # Maps identity overrides to segments
+        #
+        # @param identity_overrides [Array<Flagsmith::Engine::Identity>] Array of identity override models
+        # @return [Hash] Segments hash for identity overrides
+        def self.map_identity_overrides_to_segments(identity_overrides)
+          require 'digest'
+
+          segments = {}
+          features_to_identifiers = {}
+
+          identity_overrides.each do |identity|
+            next if identity.identity_features.nil? || !identity.identity_features.any?
+
+            # Sort features by name for consistent hashing
+            sorted_features = identity.identity_features.to_a.sort_by { |fs| fs.feature.name }
+
+            # Create override keys for hashing
+            overrides_key = sorted_features.map do |fs|
+              {
+                feature_key: fs.feature.id.to_s,
+                name: fs.feature.name,
+                enabled: fs.enabled,
+                value: fs.get_value,
+                priority: WEAKEST_PRIORITY,
+                metadata: {
+                  flagsmithId: fs.feature.id
+                }
+              }
+            end
+
+            # Create hash of the overrides to group identities with same overrides
+            overrides_hash = Digest::SHA1.hexdigest(overrides_key.to_json)
+
+            features_to_identifiers[overrides_hash] ||= { identifiers: [], overrides: overrides_key }
+            features_to_identifiers[overrides_hash][:identifiers] << identity.identifier
+          end
+
+          # Create segments for each unique set of overrides
+          features_to_identifiers.each do |overrides_hash, data|
+            segment_key = "identity_override_#{overrides_hash}"
+
+            segments[segment_key] = {
+              key: segment_key,
+              name: 'identity_override',
+              rules: [
+                {
+                  type: 'ALL',
+                  conditions: [
+                    {
+                      property: '$.identity.identifier',
+                      operator: 'IN',
+                      value: data[:identifiers].join(',')
```

Comment:
I'd rather we avoid the serde cycle for the identifiers list. The schema does support it:

```suggestion
                      value: data[:identifiers]
```